### PR TITLE
sixlowpan: minor fixes for non-ipv6 usage

### PR DESCRIPTION
**Pull request by cgundogan**
### Contribution description
When using the `sixlowpan` module without `IPv6`, then
 the compiler stumbles across some dependencies that are not met.

### Testing procedure
remove `gnrc_ipv6` from the `Makefile.dep` for `gnrc_sixlowpan` should suffice. Alternatively, I completely removed the `gnrc_sixlowpan` dependency entry from `Makefile.dep` and added `gnrc_sixlowpan_frag` to the `ccn-lite-relay` example for testing purposes.

### Issues/PRs references
–

### CODE REVIEW

File: sys/net/gnrc/network_layer/sixlowpan/gnrc_sixlowpan.c [hunk 1]
```diff
@@ -67,19 +67,18 @@ void gnrc_sixlowpan_dispatch_recv(gnrc_pktsnip_t *pkt, void *context,
 
     (void)context;
     (void)page;
-#ifdef MODULE_CCNLITE
+#ifdef MODULE_CCN_LITE
```
Comment (miri64) [inline sys/net/gnrc/network_layer/sixlowpan/gnrc_sixlowpan.c]: The code in here doesn't look CCN-lite related. How about

```suggestion
#ifndef MODULE_GNRC_IPV6
```

(or using `#ifdef MODULE_GNRC_IPV6` and turn the branches around.

File: sys/net/gnrc/network_layer/sixlowpan/gnrc_sixlowpan.c [hunk 2]
```diff
@@ -200,7 +199,11 @@ static void _receive(gnrc_pktsnip_t *pkt)
         }
 
         pkt = gnrc_pktbuf_remove_snip(pkt, sixlowpan);
+#ifdef MODULE_CCN_LITE
+        payload->type = GNRC_NETTYPE_CCN;
+#else /* MODULE_CCN_LITE */
```
Comment (miri64) [inline sys/net/gnrc/network_layer/sixlowpan/gnrc_sixlowpan.c]:
```suggestion
#elif defined(MODULE_GNRC_IPV6)
```

(and then `UNDEF` for the else branch)

**Comment by miri64** [comment]
(Compiling works)

**Comment by smlng** [comment]
Hiho, this looks easy to be moved forward and merged, right?

**Comment by miri64** [comment]
Yepp, just like to have my comment at least discussed shortly :-)

**Comment by miri64** [comment]
And go

**Pull request merged**